docs/profiling.md

```diff
@@ -0,0 +1,108 @@
+- A loader function issues requests to vHive step by step until the RPS reaches the maximum. At 
+  each step, the function issues a percentage of the maximum RPS (e.g., 5% of the maximum RPS) 
+  and profile counters. If tail latency violates 10x image unloaded service time at a step, the 
+  function returns the metric before it.
+- A single latency measurement goroutine measures the latency by injecting requests in 
+  Round-Robin and measures runtimes of requests to compute the mean latency and the 
+  90-percentile latency.
+- The tool can bind only 1 VM to a core for profiling and other VMs to a socket
+- A profiler invokes [toplev](https://github.com/andikleen/pmu-tools) to collect hardware counters.
+- A plotter plots line charts. The X-axis is the number of VMs and Y-axis is the value of a metric.
+
+A load step is divided into three phases: warm-up, profiling and cool-down. During the profile 
+period, the benchmark records the average execution time of invocations and how many invocations 
+return successfully. During the profile period, it also invokes [pmu-tools](https://github.com/andikleen/pmu-tools) 
+to profile user-defined counters followed by [the TopDown method](https://ieeexplore.ieee.org/document/6844459). 
+After the measurement finishes, completed RPS per core, average execution time and the average 
+counters are saved in the `profile.csv`.
+
+## Runtime Arguments
+```
+General:
+-warmUpTime   FLOAT  The warm up time before profiling in seconds (default 5)
+-profileTime  FLOAT  The profiling time in seconds (default 10)
+-coolDownTime FLOAT  The cool down time after profiling in seconds (default 1)
+-loadStep     FLOAT  The percentage of target RPS the benchmark loads at every step (default 5)
+-funcNames    STR    Names of the functions to benchmark, separated by comma (default helloworld)
+-profileCPU   INT    Bind only one VM to the CPU and profile the physical core only (default -1)
+-bindSocket   INT    Bind all VMs to socket number apart from the profile CPU (default -1)
+-latSamples   INT    The number of latency measurements during one profiling period (default 100)
+
+TestProfileSingleConfiguration:
+-vm           INT    The number of VMs (default 2)
+-rps          INT    The target requests per second (default 10)
+
+TestProfileIncrementConfiguration:
+-vmIncrStep   INT    The increment VM number (default 1)
+-maxVMNum     INT    The maximum VM number (default 100)
+
+Profiler:
+-l            INT    Profile level (default 1)
+-I            UINT   Print count deltas every N milliseconds (default 500)
+-nodes        STR    Include or exclude nodes (with + to add, -|^ to remove,
+                     comma separated list, wildcards allowed, 
+                     add * to include all children/siblings, 
+                     add /level to specify highest level node to match, 
+                     add ^ to match related siblings and metrics, 
+                     start with ! to only include specified nodes)
```

Comment:
Can we have more examples of the syntax? It is not easy to understand using the syntax alone.

Reply:
Done